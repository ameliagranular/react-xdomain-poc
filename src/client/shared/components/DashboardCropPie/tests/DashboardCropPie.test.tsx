import 'jest'; // necessary for types

import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';

import {
    DashboardCropPie,
    IDashboardCropPieProps
} from '../DashboardCropPie';

// Configure Enzyme for use with Jest & ES6+
configure({ adapter: new Adapter() });


const wrapper = (props: IDashboardCropPieProps = {}) => (
    <DashboardCropPie {...props} />
);

const doMount = (props: IDashboardCropPieProps = {}) => {
    return mount(wrapper(props));
};

describe('DashboardCropPie', () => {

    it('renders expected markup with no props', () => {
        expect(toJson(doMount())).toMatchSnapshot();
    });

    it('renders expected markup with pending', () => {
        expect(toJson(doMount({
            pending: true
        }))).toMatchSnapshot();
    });

    it('renders expected markup with pending and count', () => {
        expect(toJson(doMount({
            pending: true,
            count: 5
        }))).toMatchSnapshot();
    });

    it('renders expected markup with title, emptyMessage, and dataset', () => {
        expect(toJson(doMount({
            title: <span>Title</span>,
            emptyMessage: <span>No Data Found</span>,
            dataset: [
                {
                    cropTypes: [{
                        cropTypeId: '1',
                        cropTypeElement: <span>Crop 1</span>,
                        cropTypeColor: '#CCCCCC'
                    }],
                    percentage: 50,
                    amount: <span>Crop Amount 1</span>
                },
                {
                    cropTypes: [{
                        cropTypeId: '2',
                        cropTypeElement: <span>Crop 2</span>,
                        cropTypeColor: '#CCCCCC'
                    }],
                    percentage: 75,
                    amount: <span>Crop Amount 2</span>
                }
            ]
        }))).toMatchSnapshot();
    });

    it('renders expected markup with title, emptyMessage, and empty dataset', () => {
        expect(toJson(doMount({
            title: <span>Title</span>,
            emptyMessage: <span>No Data Found</span>,
            dataset: []
        }))).toMatchSnapshot();
    });

    it('renders expected markup with emptyMessage and dataset', () => {
        expect(toJson(doMount({
            emptyMessage: <span>No Data Found</span>,
            dataset: [
                {
                    cropTypes: [{
                        cropTypeId: '1',
                        cropTypeElement: <span>Crop 1</span>,
                        cropTypeColor: '#CCCCCC'
                    }],
                    percentage: 50,
                    amount: <span>Crop Amount 1</span>
                },
                {
                    cropTypes: [{
                        cropTypeId: '2',
                        cropTypeElement: <span>Crop 2</span>,
                        cropTypeColor: '#CCCCCC'
                    }],
                    percentage: 75,
                    amount: <span>Crop Amount 2</span>
                }
            ]
        }))).toMatchSnapshot();
    });
});
