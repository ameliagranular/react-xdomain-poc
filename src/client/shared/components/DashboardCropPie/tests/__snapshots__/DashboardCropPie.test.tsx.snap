// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardCropPie renders expected markup with emptyMessage and dataset 1`] = `
<DashboardCropPie
  count={5}
  dataset={
    Array [
      Object {
        "amount": <span>
          Crop Amount 1
    </span>,
        "cropTypes": Array [
          Object {
            "cropTypeColor": "#CCCCCC",
            "cropTypeElement": <span>
              Crop 1
    </span>,
            "cropTypeId": "1",
          },
        ],
        "percentage": 50,
      },
      Object {
        "amount": <span>
          Crop Amount 2
    </span>,
        "cropTypes": Array [
          Object {
            "cropTypeColor": "#CCCCCC",
            "cropTypeElement": <span>
              Crop 2
    </span>,
            "cropTypeId": "2",
          },
        ],
        "percentage": 75,
      },
    ]
  }
  emptyMessage={
    <span>
      No Data Found
    </span>
  }
  pending={false}
  title={null}
>
  <div
    className="DashboardCropPie DashboardCropPieReady"
  >
    <div
      className="DashboardCropPie_Main"
    >
      <div
        className="DashboardCropPie_Chart"
      >
        <Component
          dataset={
            Array [
              Object {
                "amount": <span>
                  Crop Amount 1
            </span>,
                "cropTypes": Array [
                  Object {
                    "cropTypeColor": "#CCCCCC",
                    "cropTypeElement": <span>
                      Crop 1
            </span>,
                    "cropTypeId": "1",
                  },
                ],
                "percentage": 50,
              },
              Object {
                "amount": <span>
                  Crop Amount 2
            </span>,
                "cropTypes": Array [
                  Object {
                    "cropTypeColor": "#CCCCCC",
                    "cropTypeElement": <span>
                      Crop 2
            </span>,
                    "cropTypeId": "2",
                  },
                ],
                "percentage": 75,
              },
            ]
          }
        >
          <div
            className="DashboardCropPie_Chart"
          >
            <div
              className="DashboardCropPie_Chart_Chart"
            >
              <div
                className="DashboardCropPie_Chart_ZingChart"
              >
                <Component
                  data={
                    Object {
                      "plot": Object {
                        "detach": false,
                        "offset-r": "3%",
                        "tooltip": Object {
                          "visible": false,
                        },
                        "value-box": Object {
                          "visible": false,
                        },
                      },
                      "scale-r": Object {
                        "ref-angle": 270,
                      },
                      "series": Array [
                        Object {
                          "background-color": "#CCCCCC",
                          "values": Array [
                            50,
                          ],
                        },
                        Object {
                          "background-color": "#CCCCCC",
                          "values": Array [
                            75,
                          ],
                        },
                        0,
                      ],
                      "type": "pie",
                    }
                  }
                  height="100%"
                  id="zingchart-5536e463-95bb-4bd9-bcff-bb7cb7909b7b"
                  width="100%"
                >
                  <div
                    className="MOCK_ZINGCHART"
                  />
                </Component>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="DashboardCropPie_Legend"
      >
        <div
          className="DashboardCropPie_Legend_Crop"
          key="1"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={null}
              cropTypes={
                Array [
                  Object {
                    "cropTypeColor": "#CCCCCC",
                    "cropTypeElement": <span>
                      Crop 1
                </span>,
                    "cropTypeId": "1",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="1"
                  style={
                    Object {
                      "background": "#CCCCCC",
                    }
                  }
                >
                  <span>
                    Crop 1
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              50
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Crop Amount 1
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="2"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={null}
              cropTypes={
                Array [
                  Object {
                    "cropTypeColor": "#CCCCCC",
                    "cropTypeElement": <span>
                      Crop 2
                </span>,
                    "cropTypeId": "2",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="2"
                  style={
                    Object {
                      "background": "#CCCCCC",
                    }
                  }
                >
                  <span>
                    Crop 2
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              75
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Crop Amount 2
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DashboardCropPie>
`;

exports[`DashboardCropPie renders expected markup with no props 1`] = `
<DashboardCropPie
  count={5}
  dataset={null}
  emptyMessage={null}
  pending={false}
  title={null}
>
  <div
    className="DashboardCropPie DashboardCropPieReady"
  >
    <div
      className="DashboardCropPie_LoneMessage"
    />
  </div>
</DashboardCropPie>
`;

exports[`DashboardCropPie renders expected markup with pending 1`] = `
<DashboardCropPie
  count={5}
  dataset={null}
  emptyMessage={null}
  pending={true}
  title={null}
>
  <div
    className="DashboardCropPie DashboardCropPiePending"
  >
    <div
      className="DashboardCropPie_Main"
    >
      <div
        className="DashboardCropPie_Chart"
      >
        <Component
          dataset={Array []}
          key="00"
        >
          <div
            className="DashboardCropPie_Chart"
          >
            <div
              className="DashboardCropPie_Chart_Chart"
            >
              <div
                className="DashboardCropPie_Chart_ZingChart"
              >
                <Component
                  data={
                    Object {
                      "plot": Object {
                        "detach": false,
                        "offset-r": "3%",
                        "tooltip": Object {
                          "visible": false,
                        },
                        "value-box": Object {
                          "visible": false,
                        },
                      },
                      "scale-r": Object {
                        "ref-angle": 270,
                      },
                      "series": Array [
                        Object {
                          "background-color": "#D6D6D6",
                          "values": Array [
                            100,
                          ],
                        },
                      ],
                      "type": "pie",
                    }
                  }
                  height="100%"
                  id="zingchart-5536e463-95bb-4bd9-bcff-bb7cb7909b7b"
                  width="100%"
                >
                  <div
                    className="MOCK_ZINGCHART"
                  />
                </Component>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="DashboardCropPie_Legend"
      >
        <div
          className="DashboardCropPie_Legend_Crop"
          key="0"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="1"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="2"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="3"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="4"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DashboardCropPie>
`;

exports[`DashboardCropPie renders expected markup with pending and count 1`] = `
<DashboardCropPie
  count={5}
  dataset={null}
  emptyMessage={null}
  pending={true}
  title={null}
>
  <div
    className="DashboardCropPie DashboardCropPiePending"
  >
    <div
      className="DashboardCropPie_Main"
    >
      <div
        className="DashboardCropPie_Chart"
      >
        <Component
          dataset={Array []}
          key="00"
        >
          <div
            className="DashboardCropPie_Chart"
          >
            <div
              className="DashboardCropPie_Chart_Chart"
            >
              <div
                className="DashboardCropPie_Chart_ZingChart"
              >
                <Component
                  data={
                    Object {
                      "plot": Object {
                        "detach": false,
                        "offset-r": "3%",
                        "tooltip": Object {
                          "visible": false,
                        },
                        "value-box": Object {
                          "visible": false,
                        },
                      },
                      "scale-r": Object {
                        "ref-angle": 270,
                      },
                      "series": Array [
                        Object {
                          "background-color": "#D6D6D6",
                          "values": Array [
                            100,
                          ],
                        },
                      ],
                      "type": "pie",
                    }
                  }
                  height="100%"
                  id="zingchart-5536e463-95bb-4bd9-bcff-bb7cb7909b7b"
                  width="100%"
                >
                  <div
                    className="MOCK_ZINGCHART"
                  />
                </Component>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="DashboardCropPie_Legend"
      >
        <div
          className="DashboardCropPie_Legend_Crop"
          key="0"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="1"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="2"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="3"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="4"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={
                Array [
                  Object {
                    "cropProductElement": <span>
                      Â 
                </span>,
                    "cropProductId": "0",
                  },
                ]
              }
              cropTypes={
                Array [
                  Object {
                    "cropTypeElement": <span>
                      Â 
                </span>,
                    "cropTypeId": "0",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="0"
                  style={
                    Object {
                      "background": "#F2F2F2",
                    }
                  }
                >
                  <span>
                    Â 
                  </span>
                </div>
                <div
                  className="CropLabel_CropProduct"
                  key="0"
                >
                  <span>
                    Â 
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              <span>
                Â 
              </span>
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Â 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DashboardCropPie>
`;

exports[`DashboardCropPie renders expected markup with title, emptyMessage, and dataset 1`] = `
<DashboardCropPie
  count={5}
  dataset={
    Array [
      Object {
        "amount": <span>
          Crop Amount 1
    </span>,
        "cropTypes": Array [
          Object {
            "cropTypeColor": "#CCCCCC",
            "cropTypeElement": <span>
              Crop 1
    </span>,
            "cropTypeId": "1",
          },
        ],
        "percentage": 50,
      },
      Object {
        "amount": <span>
          Crop Amount 2
    </span>,
        "cropTypes": Array [
          Object {
            "cropTypeColor": "#CCCCCC",
            "cropTypeElement": <span>
              Crop 2
    </span>,
            "cropTypeId": "2",
          },
        ],
        "percentage": 75,
      },
    ]
  }
  emptyMessage={
    <span>
      No Data Found
    </span>
  }
  pending={false}
  title={
    <span>
      Title
    </span>
  }
>
  <div
    className="DashboardCropPie DashboardCropPieReady"
  >
    <h2
      className="DashboardCropPie_Title"
    >
      <span>
        Title
      </span>
    </h2>
    <div
      className="DashboardCropPie_Main"
    >
      <div
        className="DashboardCropPie_Chart"
      >
        <Component
          dataset={
            Array [
              Object {
                "amount": <span>
                  Crop Amount 1
            </span>,
                "cropTypes": Array [
                  Object {
                    "cropTypeColor": "#CCCCCC",
                    "cropTypeElement": <span>
                      Crop 1
            </span>,
                    "cropTypeId": "1",
                  },
                ],
                "percentage": 50,
              },
              Object {
                "amount": <span>
                  Crop Amount 2
            </span>,
                "cropTypes": Array [
                  Object {
                    "cropTypeColor": "#CCCCCC",
                    "cropTypeElement": <span>
                      Crop 2
            </span>,
                    "cropTypeId": "2",
                  },
                ],
                "percentage": 75,
              },
            ]
          }
        >
          <div
            className="DashboardCropPie_Chart"
          >
            <div
              className="DashboardCropPie_Chart_Chart"
            >
              <div
                className="DashboardCropPie_Chart_ZingChart"
              >
                <Component
                  data={
                    Object {
                      "plot": Object {
                        "detach": false,
                        "offset-r": "3%",
                        "tooltip": Object {
                          "visible": false,
                        },
                        "value-box": Object {
                          "visible": false,
                        },
                      },
                      "scale-r": Object {
                        "ref-angle": 270,
                      },
                      "series": Array [
                        Object {
                          "background-color": "#CCCCCC",
                          "values": Array [
                            50,
                          ],
                        },
                        Object {
                          "background-color": "#CCCCCC",
                          "values": Array [
                            75,
                          ],
                        },
                        0,
                      ],
                      "type": "pie",
                    }
                  }
                  height="100%"
                  id="zingchart-5536e463-95bb-4bd9-bcff-bb7cb7909b7b"
                  width="100%"
                >
                  <div
                    className="MOCK_ZINGCHART"
                  />
                </Component>
              </div>
            </div>
          </div>
        </Component>
      </div>
      <div
        className="DashboardCropPie_Legend"
      >
        <div
          className="DashboardCropPie_Legend_Crop"
          key="1"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={null}
              cropTypes={
                Array [
                  Object {
                    "cropTypeColor": "#CCCCCC",
                    "cropTypeElement": <span>
                      Crop 1
                </span>,
                    "cropTypeId": "1",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="1"
                  style={
                    Object {
                      "background": "#CCCCCC",
                    }
                  }
                >
                  <span>
                    Crop 1
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              50
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Crop Amount 1
              </span>
            </div>
          </div>
        </div>
        <div
          className="DashboardCropPie_Legend_Crop"
          key="2"
        >
          <div
            className="DashboardCropPie_Legend_CropLabel"
          >
            <Component
              cropProducts={null}
              cropTypes={
                Array [
                  Object {
                    "cropTypeColor": "#CCCCCC",
                    "cropTypeElement": <span>
                      Crop 2
                </span>,
                    "cropTypeId": "2",
                  },
                ]
              }
            >
              <div
                className="CropLabel"
              >
                <div
                  className="CropLabel_CropType"
                  key="2"
                  style={
                    Object {
                      "background": "#CCCCCC",
                    }
                  }
                >
                  <span>
                    Crop 2
                  </span>
                </div>
              </div>
            </Component>
          </div>
          <div
            className="DashboardCropPie_Legend_CropValue"
          >
            <div
              className="DashboardCropPie_Legend_CropValue_Percentage"
            >
              75
              %
            </div>
            <div
              className="DashboardCropPie_Legend_CropValue_Amount"
            >
              <span>
                Crop Amount 2
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DashboardCropPie>
`;

exports[`DashboardCropPie renders expected markup with title, emptyMessage, and empty dataset 1`] = `
<DashboardCropPie
  count={5}
  dataset={Array []}
  emptyMessage={
    <span>
      No Data Found
    </span>
  }
  pending={false}
  title={
    <span>
      Title
    </span>
  }
>
  <div
    className="DashboardCropPie DashboardCropPieReady"
  >
    <h2
      className="DashboardCropPie_Title"
    >
      <span>
        Title
      </span>
    </h2>
    <div
      className="DashboardCropPie_LoneMessage"
    >
      <span>
        No Data Found
      </span>
    </div>
  </div>
</DashboardCropPie>
`;
